group-coordinator/src/main/java/org/apache/kafka/coordinator/group/modern/GroupSpecImpl.java

```diff
@@ -28,7 +28,7 @@
 import java.util.Objects;
 
 /**
- * The assignment specification for a consumer group.
+ * The assignment specification for a modern group.
```

Comment:
not sure whether this is necessary change. Maybe we should align the term:  "next gen", "modern", or "CONSUMER".

Reply:
Thanks for your review @chia7712. I have aligned the changes to have the common functionality as `modern`. As this spec class will be used in target assignment which will be also utilized by upcoming `share groups`. Please let me know if there is correction needed elsewhere as well.

Reply:
I don't think that we shall change the `group.coordinator.rebalance.protocols` itself. There shall remain 2 for those as `classic` and `consumer`. Package `modern` is more to share the common target assignment code between `consumer` and `share` groups. Share groups will be be specifically used for a consumption pattern where `partitions` get shared between clients to facilitate co-operative consumption.

Reply:
I raise the "naming" issue, because it is hard to distinguish the term of "consumer protocol" and "consumer instance" in introducing the "AsyncConsumer" and "CONSUMER protocol" to users. I normally give up the term used by source code and then using term "next generation" be the replacement :_

Reply:
I ll defer this question to @dajac as the naming for new consumer protocol (KIP-848) defined "consumer" as the new rebalance protocol itself. But I don't see much of relation between package names and protocol name, they can be independent.